query-engine/connectors/sql-query-connector/src/query_builder/read.rs

```diff
@@ -230,6 +236,8 @@ pub fn group_by_aggregate(
         .into_iter()
         .fold(select_query, |query, field| query.group_by(field.as_column()));
 
+    let grouped = grouped.append_trace(&Span::current());
```

Comment:
You don't _really_ need to put this into an extra line, you could do it similarly to above where you put it into the fold.